### PR TITLE
Update getGradleVersion to ignore commented out lines

### LINKED ISSUE #12357: libtxt: Text metrics are sometimes wrong (in different ways on different platforms at different sizes)
When the textScaleFactor in a RenderParagraph is not 1.0, or the font size is particular sizes (even without scaling) the Linux tests render, or at least measure, widths that are different than the same tests on Mac.  When they are off, they appear to be exactly one pixel off, and only in width.

Mac: fontSize: 10.0, textScaleFactor: 1.3 --> 13.0 wide
Linux: fontSize: 10.0, textScaleFactor: 1.3 --> 14.0 wide

Mac: fontSize: 123.0, textScaleFactor: 1.3 --> 160.0 wide
Linux: fontSize: 123.0, textScaleFactor: 1.3 --> 161.0 wide

Mac: fontSize: 100.0, textScaleFactor: 2.0 --> 200.0 wide
Linux: fontSize: 100.0, textScaleFactor: 2.0 --> 201.0 wide

BUT:
Mac: fontSize: 200.0, textScaleFactor: 2.0 --> 400.0 wide
Linux: fontSize: 200.0, textScaleFactor: 2.0 --> 400.0 wide

So it's not ALL the time, for all scales/font sizes.  This suggests some subtle difference in font rendering on one/both platforms.

Since Blink-based text rendering is being replaced, re-evaluate this bug once that has completed.

Some tests affected by this:
```
test/painting/text_painter_test.dart
test/widgets/text_test.dart
test/rendering/paragraph_test.dart
```
There is a `TODO(#12357)` in `paragraph_test.dart` that marks where the test should have the `anyOf` converted to a single value once this issue is fixed.

### PR DESCRIPTION
**Pull request by reidbaker**
https://github.com/flutter/flutter/issues/123917
Missed feedback from https://github.com/flutter/flutter/pull/123916 

- Handle commented out lines in gradle-wrapper.properties

This is 1 of 2 prs to handle missed feedback. Formatting will be done in the second whereas this one captures a weakness missed in the last pr. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/gradle_utils.dart [hunk 1]
```diff
@@ -189,8 +189,9 @@ Future<String?> getGradleVersion(
     // Expected content format (with lines above and below).
     // Version can have 2 or 3 numbers.
     // 'distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-all.zip'
+    // '^\s*' protects against commented out lines.
     final RegExp distributionUrlRegex =
-        RegExp(r'distributionUrl\s?=\s?.*\.zip');
+        RegExp(r'^\s*distributionUrl\s?=\s?.*\.zip', multiLine: true);
```
Comment (gmackall) [inline packages/flutter_tools/lib/src/android/gradle_utils.dart:194]: Question (but nit even if yes), do we want to match even if there are multiple spaces around the equals? e.g. 
```
RegExp(r'^\s*distributionUrl\s*=\s*.*\.zip', multiLine: true)
```

Comment (reidbaker) [reply packages/flutter_tools/lib/src/android/gradle_utils.dart:194]: yes I will update this.

**Review by gnprice — APPROVED** [review]
LGTM modulo the nit @gmackall spotted above.

**Review by gmackall — APPROVED** [review]
LGTM other than the mentioned nit

**Comment by reidbaker** [comment]
Frob testing is broken on something unrelated. Merging anyway.

**Pull request merged**